smtk/reddit.py

```diff
@@ -0,0 +1,144 @@
+import praw
+from smtk.utils import helpers
+import logging
+
+logging.basicConfig(format='%(asctime)s %(name)s %(levelname)s %(message)s')
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+class CollectReddit:
+
+    def __init__(self, stream=True):
+        self.auth = helpers.reddit_auth()
+        self.api = praw.Reddit(**self.auth)
+        self.stream = stream
+
+    def on_redditor(self, redditor):
+        """ Called when redditor (user) is found """
+        pass
+
+    def on_submission(self, submission):
+        """Called when submission is found"""
+        pass
+      
+    def get_redditor_comment_karma(self, names=None):
+        """"""
+        redditors = self._fetch_redditors_by_name(names=names)
+
+        if redditors: 
+            link_karma = [*map(lambda r: r.comment_karma, redditors)]
+            return link_karma
+        return None
+
+    def get_redditor_comments(self, names=None, limit=10):
+        """"""
+        redditors = self._fetch_redditors_by_name(names=names)
+
+        if redditors:
+            comments = [[comment 
+                        for comment in redditor.comments.new(limit=limit)]
+                        for redditor in redditors]
+
+            # TODO Discuss whether to return PRAW models or reduced dicts
+            # comments = [[*map(lambda c: dict(
+            #     author=c.author,
+            #     author_flair_text=c.author_flair_text,
+            #     body=c.body,
+            #     body_html=c.body_html,
+            #     edited=c.edited,
+            #     gilded=c.gilded,
+            #     link_id=c.link_id,
+            #     parent_id=c.parent_id,
+            #     score=c.score,
+            #     score_hidden=c.score_hidden,
+            #     subreddit=c.subreddit,
+            #     distinguished=c.distinguished
+            # ), user_comments)] for user_comments in comments]
+            return comments
+        return None
+
+    def get_redditor_is_gold(self, names=None):
+        """"""
+        redditors = self._fetch_redditors_by_name(names=names)
+
+        if redditors:
+            is_gold = [*map(lambda r: r.is_gold, redditors)]
+            return is_gold
+        return None
+
+    def get_redditor_is_mod(self, names=None):
```

Comment:
Example where you make it a private method and only accept 1 username. I'm not as familiar with reddit data but the core collector may not need this kind of logic unless there is a method that relies on mod status (`do_x_if_mod` etc.. If the collector is returning a praw object it can be done as the profiles are streaming.